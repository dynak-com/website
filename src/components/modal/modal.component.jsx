import React from 'react';
import styles from './styles.module.css';

const Modal = React.forwardRef((props, ref) => {
  const { content } = props;
  const modalId = content.modal_id;
  const headerTextAlignment = content.error_message
    ? 'text-justify'
    : 'text-center';
  return (
    <div
      className= 'modal show'
      id={modalId}
      ref={ref}
      tabIndex='-1'
      role='dialog'
      aria-labelledby={modalId}
      aria-hidden='true'
    >
      <div className={ `modal-dialog text-center ${ styles.modalDialog }` } role='document'>
        <div className={ `modal-content ${ styles.modalContent }` }>
          <div className='modal-header'>
            <h5 className='modal-title' id={`${modalId}Label`}>
              {content.thanks}
            </h5>
            <button
              type='button'
              className='close'
              data-dismiss='modal'
              aria-label='Close'
            >
              <span className={ `close ${ styles.closeSign }` } aria-hidden='true'>&times;</span>
            </button>
          </div>
          <div className='modal-body'>
            <h3 className={headerTextAlignment}>{content.sent}</h3>
          </div>
          <div className='text-right p-2'>
            <button type='button' className={ `btn btn-sm ${ styles.rodoCompButton }` } data-dismiss='modal'>
              {content.close}
            </button>
          </div>
        </div>
      </div>
    </div>
  );
});

export default Modal;
