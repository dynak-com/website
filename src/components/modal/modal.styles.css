.modal-dialog {
  display: table;
  height: 100%;
  width: 100%;
  vertical-align: middle;
  pointer-events: none;
  height: fit-content;
}

.modal-dialog > div {
  width: inherit;
  max-width: inherit;
  margin: 0 auto;
  color: #fff;
  background-color: #32373a;
  pointer-events: all;
}

#styled-modal-dialog {
  display: flex;
}

#rodo-comp-btn {
  color: #fff;
  background-color: #3eafad;
  transition: 0.2s;
}

#rodo-comp-btn:hover,
#rodo-comp-btn:focus,
#rodo-comp-btn:active {
  color: #ddd;
  background-color: #329390;
}

