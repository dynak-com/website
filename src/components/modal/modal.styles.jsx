import styled from 'styled-components';

export const StyledModalContainer = styled.div`
  background-color: rgba(0, 0, 0, 0);
  &.modal.show .modal-dialog,
  &.modal.fade .modal-dialog {
    /* transform: translateY(-50%); */
  }
  & > div {
    display: table;
    height: 100%;
    width: 100%;
    pointer-events: none;
  }
`;

export const StyledModalDialog = styled.div`
  display: table-cell;
  vertical-align: middle;
  pointer-events: none;
  & > div {
    width: inherit;
    max-width: inherit;
    margin: 0 auto;
    pointer-events: all;
    background-color: #32373a;
    span {
      color: #fff;
    }
    div:last-child > button {
      color: #fff;
      background-color: #3eafad;
      transition: 0.2s;
      &:hover,
      &:focus,
      &:active {
        color: #ddd;
        background-color: #329390;
      }
    }
  }
`;
