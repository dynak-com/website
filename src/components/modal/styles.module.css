.styledModalDialog {
  display: table;
  height: 100%;
  width: 100%;
  vertical-align: middle;
  pointer-events: none;
  height: fit-content;
}

.styledModalDialog .styledModalContent {
  width: inherit;
  max-width: inherit;
  margin: 0 auto;
  color: #fff;
  background-color: #32373a;
  pointer-events: all;
}

.styledModalDialog .rodoCompButton {
  color: #fff;
  background-color: #3eafad;
  transition: 0.2s;
}

.styledModalDialog .rodoCompButton:hover,
.styledModalDialog .rodoCompButton:focus,
.styledModalDialog .rodoCompButton:active {
  color: #ddd;
  background-color: #329390;
}

.styledModalDialog .closeSign {
  font-size: 1.5rem;
  color: #fff;
}

.styledModalDialog .closeSign:hover {
  color: #ddd;
}

