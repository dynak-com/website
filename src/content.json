[
  {
    "lang": "pl",
    "nav": [
      {
        "text": "nav.home",
        "address": "/"
      },
      {
        "text": "nav.about",
        "address": "/about"
      },
      {
        "text": "nav.projects",
        "address": "/projects"
      },
      {
        "text": "nav.contact",
        "address": "/contact"
      }
    ],
    "home": {
      "header": "home.header",
      "posts": [
        {
          "id": 1,
          "text": "home.posts.paragraph1",
          "headerText": "home.posts.headerText1"
        },
        {
          "id": 2,
          "text": "home.posts.paragraph2",
          "headerText": "home.posts.headerText2"
        },
        {
          "id": 3,
          "text": "home.posts.paragraph3",
          "headerText": "home.posts.headerText3"
        }
      ]
    },
    "about": {
      "header": "about.header",
      "technologies": {
        "title": "about.technologies.title",
        "content": "about.technologies.content"
      },
      "team": {
        "title": "about.team.title",
        "teamData": [
          {
            "name": "Emilia Dynak",
            "picture": "https://secure.gravatar.com/avatar/d34a1b21822840695b16113034ba740b?s=180&d=identicon",
            "position": "CEO and co-owner",
            "linkedInUrl": "https://www.linkedin.com/in/emilia-dynak-b0a3a9140/",
            "githubUrl": "https://github.com/edynak",
            "username": "edynak"
          },
          {
            "name": "Cezary Dynak",
            "picture": "https://secure.gravatar.com/avatar/47fbd06257cedcf9594bd9144f948feb?s=180&d=identicon",
            "position": "CTO and co-owner",
            "linkedInUrl": "https://www.linkedin.com/in/cdynak/",
            "githubUrl": "https://github.com/cdynak",
            "username": "cdynak"
          },
          {
            "name": "Natalia Borzyszkowska",
            "picture": "https://secure.gravatar.com/avatar/2cba91020f68596ad56766847694b292?s=180&d=identicon",
            "position": "CFO",
            "linkedInUrl": "https://www.linkedin.com/in/natalia-borzyszkowska-b7814218b/",
            "githubUrl": "https://github.com/login",
            "username": "21_smakow_natki"
          },
          {
            "name": "Aleksandra Borodziuk",
            "picture": "https://secure.gravatar.com/avatar/865d01afdfe703b4ef137693198b8500?s=180&d=identicon",
            "position": "CMO",
            "linkedInUrl": "https://www.linkedin.com/in/aleksandra-borodziuk/",
            "githubUrl": "https://github.com/login",
            "username": "Forestgirl25"
          },
          {
            "name": "Dominika Konefał",
            "picture": "https://secure.gravatar.com/avatar/ec094fb729d8e3509391628d8ba00dd8?s=180&d=identicon",
            "position": "Graphic Designer",
            "linkedInUrl": "https://www.linkedin.com/in/dominika-konefa%C5%82-028920185/",
            "githubUrl": "https://github.com/login",
            "username": "Deloi"
          },
          {
            "name": "Mikołaj Tumalewicz",
            "picture": "https://gitlab.com/uploads/-/system/user/avatar/2270247/avatar.png?width=180",
            "position": "DevOps/SysOps Engineer",
            "linkedInUrl": "https://www.linkedin.com/in/mtumalewicz/",
            "githubUrl": "https://github.com/mtumalewicz",
            "username": "mtumalewicz"
          },
          {
            "name": "Marcin Banik",
            "picture": "https://gitlab.com/uploads/-/system/user/avatar/2342806/avatar.png?s=180",
            "position": "Front-end Developer",
            "linkedInUrl": "https://www.linkedin.com/in/marcin-banik-947954142/",
            "githubUrl": "https://github.com/m-banik",
            "username": "mbanik"
          },
          {
            "name": "Bartosz Marciniec",
            "picture": "https://gitlab.com/uploads/-/system/user/avatar/4684288/avatar.png?s=180",
            "position": "Back-end Developer",
            "linkedInUrl": "https://www.linkedin.com/in/bartosz-marciniec-b0917517a/",
            "githubUrl": "https://github.com/hendas",
            "username": "hendas"
          },
          {
            "name": "Cristina Volk",
            "picture": "https://assets.gitlab-static.net/uploads/-/system/user/avatar/5111565/avatar.png?s=180",
            "position": "Web Developer",
            "linkedInUrl": "https://www.linkedin.com/in/kristina-volk-a9829517b/",
            "githubUrl": "https://github.com/CristinaVolk",
            "username": "CristinaVolk"
          }
        ]
      },
      "values": {
        "title": "about.values.title",
        "content": [
          "about.values.content.paragraph1",
          "about.values.content.paragraph2",
          "about.values.content.paragraph3"
        ]
      }
    },
    "projects": {
      "header": "projects.header",
      "posts": [
        {
          "id": 1,
          "text": "projects.posts.text1",
          "headerText": "dynak.io"
        },
        {
          "id": 2,
          "text": "projects.posts.text2",
          "headerText": "warehouse-app"
        },
        {
          "id": 3,
          "text": "projects.posts.text3",
          "headerText": "projects.posts.headerText3"
        }
      ]
    },
    "contact": {
      "header": "contact.header",
      "socials": "contact.socials",
      "form": {
        "subject": "contact.form.subject",
        "email": "contact.form.email",
        "message": "contact.form.message",
        "contactAgreement": "contact.form.contactAgreement",
        "privacyPolicyAgreement": "contact.form.privacyPolicyAgreement",
        "policyPrivacy": "contact.form.policyPrivacy",
        "send": "contact.form.send"
      }
    },
    "modal": {
      "modal_ok": {
        "modal_id": "modal_ok",
        "error_message": false,
        "thanks": "modal_ok.thanks",
        "sent": "modal_ok.sent",
        "close": "modal_ok.close"
      },
      "modal_error": {
        "modal_id": "modal_error",
        "error_message": true,
        "thanks": "modal_error.thanks",
        "sent": "modal_error.sent",
        "close": "modal_error.close"
      },
      "modal_privacy_policy": {
        "modal_id": "modal_privacy_policy",
        "error_message": null,
        "thanks": "modal_privacy_policy.thanks",
        "sent": "modal_privacy_policy.sent",
        "close": "modal_privacy_policy.close",
        "rodoContent": {
          "header": "rodoContent.header",
          "sentence1": "rodoContent.sentence1",
          "website": "rodoContent.website",
          "sentence2": "rodoContent.sentence2",
          "generalProvisions": "rodoContent.generalProvisions",
          "text1": "rodoContent.text1",
          "text2": "rodoContent.text2",
          "text3": "rodoContent.text3",
          "definitions": "rodoContent.definitions",
          "administrator": "rodoContent.administrator",
          "user": "rodoContent.user",
          "personalData": "rodoContent.personalData",
          "paragraph1": "rodoContent.paragraph1",
          "linkedIn": "rodoContent.linkedIn",
          "paragraph2": "rodoContent.paragraph2",
          "paragraph3": "rodoContent.paragraph3",
          "paragraph4": "rodoContent.paragraph4",
          "paragraph5": "rodoContent.paragraph5",
          "paragraph6": "rodoContent.paragraph6",
          "paragraph7": "rodoContent.paragraph7",
          "paragraph8": "rodoContent.paragraph8",
          "webLinkedIn": "rodoContent.webLinkedIn",
          "github": "rodoContent.github",
          "gitlab": "rodoContent.gitlab",
          "paragraph9": "rodoContent.paragraph9",
          "paragraph10": "rodoContent.paragraph10",
          "paragraph11": "rodoContent.paragraph11",
          "paragraph12": "rodoContent.paragraph12",
          "paragraph13": "rodoContent.paragraph13",
          "paragraph14": "rodoContent.paragraph14",
          "paragraph15": "rodoContent.paragraph15",
          "paragraph16": "rodoContent.paragraph16",
          "paragraph17": "rodoContent.paragraph17",
          "paragraph18": "rodoContent.paragraph18",
          "paragraph19": "rodoContent.paragraph19",
          "paragraph20": "rodoContent.paragraph20",
          "paragraph21": "rodoContent.paragraph21",
          "paragraph22": "rodoContent.paragraph22",
          "paragraph23": "rodoContent.paragraph23",
          "forms": "rodoContent.forms",
          "paragraph24": "rodoContent.paragraph24",
          "paragraph25": "rodoContent.paragraph25",
          "technologies": "rodoContent.technologies",
          "Administrator": "rodoContent.Administrator",
          "not": "rodoContent.not",
          "paragraph26": "rodoContent.paragraph26",
          "cookies": "rodoContent.cookies",
          "part1": "rodoContent.part1",
          "part2": "rodoContent.part2",
          "part3": "rodoContent.part3",
          "part4": "rodoContent.part4",
          "created": "rodoContent.created"
        }
      }
    },
    "error": "error",
    "footer": {
      "rodo": "footer.rodo",
      "dynak": "footer.dynak"
    }
  }
]