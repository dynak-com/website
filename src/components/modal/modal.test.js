import React from 'react';
import { shallow } from 'enzyme';
import { Modal } from './modal.component';
import content from '../../content.json';

describe('Modal component', () => {
    const [contentData] = content;
    const { modal } = contentData;

    it('renders correctly according to Snapshot', () => {});
});
