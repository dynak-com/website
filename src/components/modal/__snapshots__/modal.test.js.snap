// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders Invalid Modal component according to snapshot 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="modal_error"
  className="modal show"
  id="modal_error"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog text-center modalDialog"
    role="document"
  >
    <div
      className="modal-content modalContent"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="modal_errorLabel"
        >
          Niestety :(
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
            className="close closeSign"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <h3
          className="text-justify"
        >
          Twoja wiadomość nie została wysłana. Nastąpił błąd po stronie serwera. Spróbuj ponownie później. Przepraszamy!
        </h3>
      </div>
      <div
        className="text-right p-2"
      >
        <button
          className="btn btn-sm rodoCompButton"
          data-dismiss="modal"
          type="button"
        >
          Zamknij
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal component renders correctModal component according to snapshot 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="modal_ok"
  className="modal show"
  id="modal_ok"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog text-center modalDialog"
    role="document"
  >
    <div
      className="modal-content modalContent"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="modal_okLabel"
        >
          Dziękujemy :)
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
            className="close closeSign"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <h3
          className="text-center"
        >
          Twoja wiadomość została wysłana.
        </h3>
      </div>
      <div
        className="text-right p-2"
      >
        <button
          className="btn btn-sm rodoCompButton"
          data-dismiss="modal"
          type="button"
        >
          Zamknij
        </button>
      </div>
    </div>
  </div>
</div>
`;
